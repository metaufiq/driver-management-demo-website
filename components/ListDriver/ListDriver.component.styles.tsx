import styled from "styled-components";

export const SpinnerContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 5rem;
`

export const ErrorContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  flex-direction: column;
  color: gray;
  gap: 1rem;
  padding: 4rem;
`

export const DataContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(5, minmax(0, 1fr));
  gap: 1rem;
  overflow-x: scroll;
  background-color: #fafafa;
  padding: 2rem 0;
  @media (max-width: 1430px) {
    grid-template-columns: repeat(4, minmax(0, 1fr));
  }
  @media (max-width: 1280px) {
    grid-template-columns: repeat(3, minmax(0, 1fr));
  }
  @media (max-width: 1024px) {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }
  @media (max-width: 768px) {
    grid-template-columns: repeat(1, minmax(0, 1fr));
  }
`