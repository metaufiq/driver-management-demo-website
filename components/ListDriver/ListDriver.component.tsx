import { Component } from "../../index.types"
import { PrimaryButton } from "../Button"
import DriverCard from "../DriverCard"
import Spinner from "../Spinner"
import { DataContainer, ErrorContainer, SpinnerContainer } from "./ListDriver.component.styles"
import { Props } from "./ListDriver.component.types"

const ListDriver:Component<Props> = ({isLoading, isError, data, refetch})=>{
  if (isLoading) {
   return(
    <SpinnerContainer>
      <Spinner size='4rem'/>
    </SpinnerContainer>
   ) 
  }

  if (isError) {
    return(
      <ErrorContainer>
        <p>Terjadi Kesalahan Saat pengambilan Data Driver.</p>
        <PrimaryButton onClick={refetch}>Coba Lagi</PrimaryButton>
      </ErrorContainer>
    )
  }

  return(
    <DataContainer>
      {data?.map(user=>(<DriverCard user={user} key={user.username}/>))}
    </DataContainer>
  )
}

export default ListDriver