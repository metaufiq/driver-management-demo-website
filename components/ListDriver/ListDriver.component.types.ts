import { Users } from "../../index.types"


export type Props = {
  data: Users,
  isLoading?: boolean,
  isError?: boolean,
  refetch?: VoidFunction
}