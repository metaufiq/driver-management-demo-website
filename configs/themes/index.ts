const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };