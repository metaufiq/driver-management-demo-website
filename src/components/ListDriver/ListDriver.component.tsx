import { ReactElement } from "react"
import { Component } from "../../../index.types"
import Button from "../Button"
import DriverCard from "../DriverCard"
import Spinner from "../Spinner"
import { DataContainer, ErrorContainer, SpinnerContainer } from "./ListDriver.component.styles"
import { Props } from "./ListDriver.component.types"

/**
 * List Driver components
 * @param {Props} props component props 
 * @returns {ReactElement} List Driver components
 */
const ListDriver:Component<Props> = (
  {isLoading, isError, data, refetch}: Props
): ReactElement=>{
  if (isLoading) {
   return(
    <SpinnerContainer>
      <Spinner size='4rem'/>
    </SpinnerContainer>
   ) 
  }

  if (isError) {
    return(
      <ErrorContainer>
        <p>Terjadi Kesalahan Saat pengambilan Data Driver.</p>
        <Button onClick={refetch}>Coba Lagi</Button>
      </ErrorContainer>
    )
  }

  return(
    <DataContainer>
      {data?.map(user=>(<DriverCard user={user} key={user.username}/>))}
    </DataContainer>
  )
}

export default ListDriver