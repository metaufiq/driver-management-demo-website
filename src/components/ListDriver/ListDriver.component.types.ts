import { Users } from "@dmd/types"


export type Props = {
  data: Users,
  isLoading?: boolean,
  isError?: boolean,
  refetch?: VoidFunction
}