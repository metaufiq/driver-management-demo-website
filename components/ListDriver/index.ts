import ListDriver from "./ListDriver.component";

export default ListDriver