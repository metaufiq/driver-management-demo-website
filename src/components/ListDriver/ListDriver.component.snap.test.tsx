import { assertSnapshots } from "../../test";
import { Configs } from "../../test/index.types";
import ListDriver from ".";
import { Props } from "./ListDriver.component.types";
import { themes } from "../../configs/themes";
import { Users } from "../../../index.types";

describe('ListDriverSnapshot', () => {

  const data:Users = [{
    dob: '12/10/1998',
    email: 'email@email.com',
    firstName: 'firstName',
    lastName: 'lastName',
    phone: '098123123123',
    username: 'userName',
    picture: {
      large: 'large_picture_url',
      medium: 'medium_picture_url',
      thumbnail: 'thumbnail_picture_url',
    }
   }]

  const configs: Configs<Props> = [
    {
      desc: 'should render normal state list driver',
      props:{
        data, 
      },
      theme: themes,
      useTheme: true
    },
    {
      desc: 'should render loading state list driver',
      props:{
        data: [],
        isLoading: true 
      },
      theme: themes,
      useTheme: true
    },
    {
      desc: 'should render error state list driver',
      props:{
        data: [],
        isError: true 
      },
      theme: themes,
      useTheme: true
    },
  ];

  assertSnapshots(ListDriver, configs);
});
