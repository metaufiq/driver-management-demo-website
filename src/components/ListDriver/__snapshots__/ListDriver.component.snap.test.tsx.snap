// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDriverSnapshot should render error state list driver 1`] = `
<div
  className="sc-ksZaOG cNQmIU"
>
  <p>
    Terjadi Kesalahan Saat pengambilan Data Driver.
  </p>
  <button
    className="sc-bczRLJ iyjxpd"
  >
    Coba Lagi
  </button>
</div>
`;

exports[`ListDriverSnapshot should render loading state list driver 1`] = `
<div
  className="sc-breuTD gNnOMM"
>
  <div
    className="sc-evZas cVQZXf"
    size="4rem"
  />
</div>
`;

exports[`ListDriverSnapshot should render normal state list driver 1`] = `
<div
  className="sc-hAZoDl jsEuU"
>
  <div
    className="sc-gsnTZi byoYPo"
  >
    <div
      className="sc-hKMtZM dfTmvn"
    >
      <div>
        <p
          className="sc-dkzDqf sc-eCYdqJ bkDjgV geixOC"
        >
          Driver ID
        </p>
        <p
          className="sc-jSMfEi eQOOWD"
        >
          userName
        </p>
      </div>
      <div
        className="sc-gKXOVf kSkZUi"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 10a2 2 0 11-4 0 2 2 0 014 0zM12 10a2 2 0 11-4 0 2 2 0 014 0zM16 12a2 2 0 100-4 2 2 0 000 4z"
          />
        </svg>
      </div>
    </div>
    <div
      className="sc-papXJ hIXhCT"
    >
      <div
        className="sc-crXcEl eDMxcY"
      />
      <div
        className="sc-jqUVSM epcwnv"
      >
        <div
          className="sc-kDDrLX UBIxm"
        >
          <p
            className="sc-dkzDqf sc-iBkjds bkDjgV fDTFxa"
          >
            Nama Driver
          </p>
          <p
            className="sc-ftvSup gjtXSY"
          >
            firstName
            , 
            lastName
          </p>
        </div>
        <div
          className="sc-kDDrLX UBIxm"
        >
          <p
            className="sc-dkzDqf sc-iBkjds bkDjgV fDTFxa"
          >
            Telepon
          </p>
          <p
            className="sc-ftvSup gjtXSY"
          >
            098123123123
          </p>
        </div>
        <div
          className="sc-iqcoie fmUYHX"
        >
          <div
            className="sc-kDDrLX UBIxm"
          >
            <p
              className="sc-dkzDqf sc-iBkjds bkDjgV fDTFxa"
            >
              Email
            </p>
            <p
              className="sc-ftvSup gjtXSY"
            >
              email@email.com
            </p>
          </div>
          <div
            className="sc-kDDrLX UBIxm"
          >
            <p
              className="sc-dkzDqf sc-iBkjds bkDjgV fDTFxa"
            >
              Tanggal Lahir
            </p>
            <p
              className="sc-ftvSup gjtXSY"
            >
              12/10/1998
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
